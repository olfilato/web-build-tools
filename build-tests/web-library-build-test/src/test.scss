body {
  background: red;
}

.foo {
  border: 1px solid red;
}
