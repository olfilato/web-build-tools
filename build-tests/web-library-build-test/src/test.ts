// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import test from './test.scss';
import testFunction from './preCopyTest';

/** @public */
export function log(message: string): void {
  console.log(message);
}

/** @public */
export function add(num1: number, num2: number): number {
  return num1 + num2;
}

/** @public */
export function logClass(): void {
  console.log(test.foo);
}

testFunction();
