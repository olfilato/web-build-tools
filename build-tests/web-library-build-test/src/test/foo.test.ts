// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
// See LICENSE in the project root for license information.

import { assert } from 'chai';
import { add } from '../test';

describe('foo test', () => {
  it('can assert using chai', () => {
    assert(add(1, 2) === 3);
  });

  it('can assert using jest', () => {
    expect(add(1, 2)).toBe(3);
  });

  it('can test snapshot', () => {
    expect(add(2 / 3, 9 / 11)).toMatchSnapshot();
  });
});
