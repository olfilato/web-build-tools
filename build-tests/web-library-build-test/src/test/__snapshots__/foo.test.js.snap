// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`foo test can test snapshot 1`] = `1.4848484848484849`;
